### PR TITLE
feat(nm): added network failover 'route-metric'

### PR DESCRIPTION
**Pull request by marcellorinaldo**
This PR adds the property `route-metric` when the interface is enabled as WAN, so that _NetworkManager_ can implement the network failover when the lowest priority interface is not able to connect.

Reference on [network failover](https://www.digi.com/resources/documentation/digidocs/90002285/reference/yocto/r_network_failover.htm?TocPath=Digi%20Embedded%20Yocto%7CSystem%20development%7CSoftware%20extensions%7C_____3).

**Related Issue:** N/A.

**Description of the solution adopted:**

- Added new metric.
- Removed piece of code that disables the old WAN interface when a new is activated as WAN.
- Removed snapshot take when old WAN interfaces is disabled.
- Added validation on priority input values.

**Screenshots:**

image

**Any side note on the changes made:** N/A.

### CODE REVIEW

**Comment by MMaiero** [comment]
@marcellorinaldo Can you please check as tests are failing?

**Comment by mattdibi** [comment]
@marcellorinaldo We merged https://github.com/eclipse/kura/pull/4380. You can now rebase and add the unit test for the changes contained in this PR

File: kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMSettingsConverter.java [hunk 1]
```diff
@@ -100,10 +100,17 @@ public static Map<String, Variant<?>> buildIpv4Settings(NetworkProperties props,
                 settings.put("dns", new Variant<>(convertIp4(dnsServers.get()), "au"));
                 settings.put("ignore-auto-dns", new Variant<>(true));
             }
+            
             Optional<String> gateway = props.getOpt(String.class, "net.interface.%s.config.ip4.gateway", iface);
             if (gateway.isPresent()) {
                 settings.put("gateway", new Variant<>(gateway.get()));
             }
+
+            Optional<Integer> wanPriority = props.getOpt(Integer.class, "net.interface.%s.config.ip4.wan.priority",
```
Comment (mattdibi) [inline kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMSettingsConverter.java:109]:
```suggestion
            Optional<Long> wanPriority = props.getOpt(Long.class, "net.interface.%s.config.ip4.wan.priority",
```

**Review by mattdibi — APPROVED** [review]
LGTM

Tested on freshly installed RPi, `route-metric` is correctly set on the desired connection.

image


image

**Comment by MMaiero** [comment]
Tests need to be addressed before being ok to merge

**Comment by marcellorinaldo** [comment]
Screenshot 2023-02-17 at 12 29 13

PR fails due to CI not being able to access Kura project on Sonar

**Pull request merged**